### PR TITLE
Move some per-tile tests to maplints.

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do
This PR moves some per-tile tests into maplints. Per-tile tests are almost certainly slower on measure, and I prefer the way @Contrabang is using them, for verifying tile state after map load. Maplints should be used for map pre-load checks; anything that doesn't need atoms being initialized.

Some details:
- there were two checks involving checking for duplicate cables on tiles, one just for nodes and one that compared each cable's d1 and d2. These are both already handled by `stacked_matching_cables.yml`.
- that there was a check for structures on centcomm areas that never ran because centcomm was never passed as a z-level, and it fails because the windows surrounding the holodeck scenes are on space areas, and I'm not touching that part of the map.
## Why It's Good For The Game
Almost certainly faster, even if only in CI compilation. Less code.
## Testing
Ran em

<hr>

### Declaration

- [X] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog
NPFC

### CODE REVIEW

File: tools/maplint/lints/invalid_space_neighbors.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/area/space:
+  banned_neighbors:
+  - /obj/machinery/door/airlock
+=/area/space:
```
Comment (Contrabang) [inline tools/maplint/lints/invalid_space_neighbors.yml:4]: any reason not to move this up to all subtypes?

File: code/modules/unit_tests/map_tests.dm [hunk 2]
```diff
@@ -145,16 +81,6 @@
...
 
-/datum/map_per_tile_test/duplicate_cable_check
-
-/datum/map_per_tile_test/duplicate_cable_check/CheckTile(turf/T)
-	var/obj/structure/cable/cable = locate() in T.contents
-	for(var/obj/structure/cable/other_cable in T.contents)
-		if(cable == other_cable)
-			continue // same object, continue
-		if(cable.d1 == other_cable.d1 && cable.d2 == other_cable.d2)
-			Fail(T, "tile has duplicated cables.")
```
Comment (Contrabang) [inline code/modules/unit_tests/map_tests.dm:156]: accidental removal? These need to check that they have the exact same direction so a normal maplint wont work here

File: code/modules/unit_tests/map_tests.dm [hunk 3]
```diff
@@ -19,48 +19,6 @@
...
-
-/datum/map_per_tile_test/cable_node_checker/CheckTile(turf/T)
-	var/center_nodes = 0
-
-	for(var/obj/structure/cable/cable in T.contents)
-		if(cable.d1 == 0 || cable.d2 == 0)
-			center_nodes++
-
-	if(center_nodes > 1)
-		Fail(T, "tile has multiple center cable nodes")
```
Comment (Contrabang) [inline code/modules/unit_tests/map_tests.dm:62]: accidental deletion here too?

File: tools/maplint/lints/unary_vents.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/obj/machinery/atmospherics/unary:
+  banned_neighbors:
+  - /obj/machinery/atmospherics/pipe/manifold/hidden/supply
+  - /obj/machinery/atmospherics/pipe/simple/hidden/scrubbers
```
Comment (Contrabang) [inline tools/maplint/lints/unary_vents.yml:4]: this should probably be all pipes

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/modules/unit_tests/map_tests.dm (hunk 2 above)
Comment (warriorstar-orion) [reply code/modules/unit_tests/map_tests.dm:156]: This and the other cable test are both already covered by `stacked_matching_cables.yml`:
```yaml
/obj/structure/cable:
  banned_neighbors:
    /obj/structure/cable:
      identical: true
```

This prevents all cables with the same varedits. So if two cables have the same icon-state, they'll fail this lint.

File: code/modules/unit_tests/map_tests.dm (hunk 3 above)
Comment (warriorstar-orion) [reply code/modules/unit_tests/map_tests.dm:62]: See above.

File: tools/maplint/lints/invalid_space_neighbors.yml (hunk 1 above)
Comment (warriorstar-orion) [reply tools/maplint/lints/invalid_space_neighbors.yml:4]: Move what up?

File: tools/maplint/lints/unary_vents.yml (hunk 4 above)
Comment (warriorstar-orion) [reply tools/maplint/lints/unary_vents.yml:4]: Just porting the tests for now, can tighten them up later

File: tools/maplint/lints/invalid_space_neighbors.yml (hunk 1 above)
Comment (Contrabang) [reply tools/maplint/lints/invalid_space_neighbors.yml:4]: the structure check to all subtypes of space

File: code/modules/unit_tests/map_tests.dm (hunk 2 above)
Comment (Contrabang) [reply code/modules/unit_tests/map_tests.dm:156]: oh, amazing

File: code/modules/unit_tests/map_tests.dm (hunk 3 above)
Comment (Contrabang) [reply code/modules/unit_tests/map_tests.dm:62]: Non-identical nodes can trigger this, for example 1 cable has `d1 = 1, d2 = 0` and a second cable has `d1 = 0, d2 = 4`.

File: tools/maplint/lints/invalid_space_neighbors.yml (hunk 1 above)
Comment (warriorstar-orion) [reply tools/maplint/lints/invalid_space_neighbors.yml:4]: I'll add explicit checks for `/area/space/no_teleport` and `/area/space/nearstation` (the other two relevant types) but I can't use this for `/area/space/centcomm` because there are windows surrounding the holodeck scenes that I'm not touching

File: code/modules/unit_tests/map_tests.dm (hunk 3 above)
Comment (warriorstar-orion) [reply code/modules/unit_tests/map_tests.dm:62]: Ugh, what a pain, I can't express that in maplint, though maybe I can add it at some point (testing to see if both icon_states have a zero in them somewhere)

File: tools/maplint/lints/invalid_space_neighbors.yml (hunk 1 above)
Comment (warriorstar-orion) [reply tools/maplint/lints/invalid_space_neighbors.yml:6]: Wait, none of those work, that's why I did that, fuck

**Review by Contrabang — APPROVED** [review]
LGTM

**Review by Burzah — APPROVED** [review]
